plugins/hypervisors/vmware/src/main/java/com/cloud/hypervisor/guru/VMwareGuru.java

```diff
@@ -1104,13 +1124,22 @@ public List<Command> finalizeMigrate(VirtualMachine vm, Map<Volume, StoragePool>
             if (pool.getClusterId() != null) {
                 poolClusterId = pool.getClusterId();
             }
+            if (volume.getVolumeType().equals(Volume.Type.ROOT) && pool.isLocal()) {
+                isLocalStorageMigration = true;
```

Comment:
Better to get the host guid for local storage here, using the pool id (no need to pass the entire volume/pool map).

`getHostGuidForLocalStorage(Long poolId)`

Reply:
I've intentionally kept it out of this loop to retrieve the target host guid for localstorage and shared storage together.